From 4ad311ebb6c1d98dae302617a5a9c4dd66219179 Mon Sep 17 00:00:00 2001
From: Daniel Hoffend <dh@dotlan.net>
Date: Tue, 26 Nov 2019 09:22:47 +0100
Subject: [PATCH 2/3] Fixed T5653 and support domain_edit in hosted mode

Summary:
This patch fixed the domain_edit api call in hosted mode. The Bug is
described in T5653

Reviewers: #web_administration_panel_developers, vanmeeuwen

Reviewed By: #web_administration_panel_developers, vanmeeuwen

Subscribers: vanmeeuwen

Tags: #web_administration_panel

Differential Revision: https://git.kolab.org/D835
---
 lib/Auth/LDAP.php | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index 4bf2a8f..aac6df8 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -396,6 +396,11 @@ class LDAP extends Net_LDAP3 {
             return false;
         }
 
+        // remove invalid/hidden attributes
+        if (isset($attributes['domainrelatedobject_only'])) {
+            unset($attributes['domainrelatedobject_only']);
+        }
+
         $domain_dn = key($domain);
 
         // We should start throwing stuff over the fence here.
-- 
2.20.1

